### ISSUE TITLE
Text segmentation is slow

### ISSUE THREAD

**Issue by waybackarchiver** [zoomio/tagify#23]
When processing large amounts of text, the three locations listed below perform poorly, perhaps more optimization is needed. 

https://github.com/zoomio/tagify/blob/15b8055025ca47ffd852c14a47a2dc94d0a0da55/processor/html/html_processor.go#L374-L413

https://github.com/zoomio/tagify/blob/15b8055025ca47ffd852c14a47a2dc94d0a0da55/processor/text/text_processor.go#L46-L71

https://github.com/zoomio/tagify/blob/15b8055025ca47ffd852c14a47a2dc94d0a0da55/processor/md/md_processor.go#L245-L276

**Comment by smeshkov** [comment]
Will take a look, thanks for reporting @waybackarchiver 👍

**Comment by smeshkov** [comment]
@waybackarchiver did you notice it for the Chinese & Japanese languages only, for all languages or maybe all languages except Chinese & Japanese ? Just trying to grasp the scope atm.

**Comment by waybackarchiver** [comment]
> @waybackarchiver did you notice it for the Chinese & Japanese languages only, for all languages or maybe all languages except Chinese & Japanese ? Just trying to grasp the scope atm.

I only tested in Chinese.

**Comment by smeshkov** [comment]
@waybackarchiver yeah looks like it is the usage of `go-ego/gse`, I use it only for Chinese and Japanese languages:
```
BenchmarkParseHTML-10            	          20474	         49361 ns/op
BenchmarkParseHTML-10            	          22647	         45415 ns/op
BenchmarkParseHTML-10            	          17845	         56261 ns/op
BenchmarkParseHTML-10            	          16270	         61920 ns/op
BenchmarkParseHTML-10            	          10144	        99633 ns/op
BenchmarkParseHTML_chinese-10    	     259	   4052756 ns/op
BenchmarkParseHTML_chinese-10    	     307	   3333507 ns/op
BenchmarkParseHTML_chinese-10    	     350	   3151525 ns/op
BenchmarkParseHTML_chinese-10    	     319	   3433867 ns/op
BenchmarkParseHTML_chinese-10    	     331	   3253833 ns/op
```

I'll dig deeper to see what options we have.
